src/commands/deleteResourceGroup/v2/deleteResourceGroupV2.ts

```diff
@@ -3,72 +3,110 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizard, IActionContext, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import { ResourceGroup } from '@azure/arm-resources';
+import { uiUtils } from '@microsoft/vscode-azext-azureutils';
+import { AzureWizard, IActionContext, IAzureQuickPickItem, nonNullProp, subscriptionExperience, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import { AzureResource, AzureSubscription } from '@microsoft/vscode-azext-utils/hostapi.v2';
+import { DefaultAzureResourceProvider } from '../../../api/DefaultAzureResourceProvider';
 import { ext } from '../../../extensionVariables';
 import { GroupingItem } from '../../../tree/azure/GroupingItem';
 import { createActivityContext } from '../../../utils/activityUtils';
+import { createResourceClient } from '../../../utils/azureClients';
 import { localize } from '../../../utils/localize';
 import { settingUtils } from '../../../utils/settingUtils';
+import { createSubscriptionContext } from '../../../utils/v2/credentialsUtils';
 import { DeleteResourceGroupContext } from '../DeleteResourceGroupContext';
 import { DeleteResourceGroupStep } from '../DeleteResourceGroupStep';
 
 export async function deleteResourceGroupV2(context: IActionContext, primaryNode?: GroupingItem, selectedNodes?: GroupingItem[]): Promise<void> {
+
+    // unset nodes that are not resource groups
+    selectedNodes = selectedNodes?.filter(n => !!n.resourceGroup);
+    if (!primaryNode?.resourceGroup) {
```

Comment:
Is this property unique to resource group resources? I'm pretty sure it exists on other resources. Shouldn't we check the `type` property?

Reply:
Good catch. I don't think it currently exists on items other than the resource groups `GroupingItems`, however I could see this causing a bug in the future if it was added to another item.

I'll do an `instanceof` check as well. I really like avoiding using `instanceof` because it can cause other issues, but it's necessary here for now.

Reply:
Why not just check for this `"type": "Microsoft.Resources/resourceGroups"?

Reply:
The object is a `GroupingItem`, not an `AzureResource`.